src/Collections/Maps/Maps.dfy

```diff
@@ -127,4 +164,33 @@ module {:options "-functionSyntax:4"} Maps {
     forall x, x' {:trigger m[x], m[x']} :: x in m && x' in m && start <= x <= x' ==> m[x] <= m[x']
   }
 
+  /* Maps an injective function over the keys of a map, retaining the values. */
+  function {:opaque} MapKeys<X(!new), Y, X'>(m: map<X, Y>, f: X --> X'): (m': map<X', Y>)
```

Comment:
Unfortunate that `Seq.Map` has the parameters in the other order. This is something we should align on better in the `src/dafny/...` copy of all the collections code.

Reply:
Since these are new APIs I'll just switch them around now.